packages/cli/src/util/dev/server.ts

```diff
@@ -1351,6 +1350,25 @@ export default class DevServer {
     return false;
   };
 
+  runDevMiddleware = async (
+    req: http.IncomingMessage,
+    res: http.ServerResponse
+  ) => {
+    const { devMiddlewarePlugins } = await loadCliPlugins(
+      this.cwd,
+      this.output
+    );
+    if (devMiddlewarePlugins.length) {
```

Comment:
If we're only expecting one, then `loadCliPlugins` should probably only return one or zero